### PR TITLE
test: adding integ tests for fss triggered by local deployment

### PR DESCRIPTION
**Pull request by J-C-L**
**Issue #, if available:**

**Description of changes:**
adding integ tests for fss triggered by local deployment

**Why is this change necessary:**

**How was this change tested:**
- [ ] Updated or added new unit tests.
- [X] Updated or added new integration tests.
- [ ] Updated or added new end-to-end tests.
- [ ] If my code makes a remote network call, it was tested with a proxy.

**Any additional information or context required to review the change:**

**Documentation Checklist:**
 - [ ] Updated the README if applicable.

**Compatibility Checklist:**
- [ ] I confirm that the change is backwards compatible.
- [ ] Any modification or deletion of public interfaces does not impact other plugin components.
- [ ] For external library version updates, I have reviewed its change logs and Nucleus does not consume 
  any deprecated method or type.

Refer to [Compatibility Guidelines](/COMPATIBILITY.md) for more information.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by junfuchen99 — DISMISSED** [review]

**Review by indougnito — APPROVED** [review]

File: src/integrationtests/java/com/aws/greengrass/integrationtests/status/EventFleetStatusServiceTest.java [hunk 1]
```diff
@@ -233,8 +233,100 @@ void GIVEN_jobs_deployment_WHEN_deployment_finishes_THEN_status_is_uploaded_to_c
...
+    void GIVEN_local_deployment_WHEN_deployment_finishes_THEN_status_is_uploaded_to_cloud() throws Exception {
+        CountDownLatch fssPublishLatch = new CountDownLatch(1);
+        logListener = eslm -> {
+            if (eslm.getEventType() != null
+                    && eslm.getEventType().equals("fss-status-update-published")
+                    && eslm.getMessage().contains("Status update published to FSS")) {
+                fssPublishLatch.countDown();
+            }
+        };
+        Slf4jLogAdapter.addGlobalListener(logListener);
```
Comment (rbattle) [inline src/integrationtests/java/com/aws/greengrass/integrationtests/status/EventFleetStatusServiceTest.java:245]: wrap this in a try/catch so the logListener gets removed on error

You can do:
```
 try (AutoCloseable ignored = TestUtils.createCloseableLogListener(listener)) {
   // test
 }
 ```

File: src/integrationtests/java/com/aws/greengrass/integrationtests/status/EventFleetStatusServiceTest.java [hunk 2]
```diff
@@ -233,8 +233,100 @@ void GIVEN_jobs_deployment_WHEN_deployment_finishes_THEN_status_is_uploaded_to_c
...
+                .requestTimestamp(System.currentTimeMillis())
+                .build();
+        submitLocalDocument(secondRequest);
+
+        assertTrue(fssPublishLatch.await(180, TimeUnit.SECONDS));
+        verify(mqttClient, atLeastOnce()).publish(captor.capture());
+        List<PublishRequest> prs = captor.getAllValues();
+
+        // Get the last FSS publish request which should have no new component info, since no component statuses have
+        // changed
```
Comment (rbattle) [inline src/integrationtests/java/com/aws/greengrass/integrationtests/status/EventFleetStatusServiceTest.java:315]: if the component had finished, would this have published a change/restarted the component? I'm assuming not - is this correct?

File: src/integrationtests/java/com/aws/greengrass/integrationtests/status/EventFleetStatusServiceTest.java (hunk 1 above)
Comment (J-C-L) [reply src/integrationtests/java/com/aws/greengrass/integrationtests/status/EventFleetStatusServiceTest.java:245]: Good call. Changed for the new tests and the pre-existing ones.

File: src/integrationtests/java/com/aws/greengrass/integrationtests/status/EventFleetStatusServiceTest.java (hunk 2 above)
Comment (J-C-L) [reply src/integrationtests/java/com/aws/greengrass/integrationtests/status/EventFleetStatusServiceTest.java:315]: No. Shouldn't change the pre-existing behavior in that way.

**Review by junfuchen99 — APPROVED** [review]

**Review by rbattle — APPROVED** [review]

**Pull request merged**